From f587cb900b3100f31d379f3f9e1b60521c5ebc69 Mon Sep 17 00:00:00 2001
From: Daniel Baumann <mail@daniel-baumann.ch>
Date: Mon, 10 Nov 2014 14:18:04 +0100
Subject: [PATCH] Dropping usage of /etc/fstab.d since util-linux removed it,
 thanks to Evgeni Golov <evgeni+git@golov.de> for reporting it.

---
 components/3020-swap     | 8 ++++----
 components/9990-fstab.sh | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/components/3020-swap b/components/3020-swap
index a1bcdbe..345b9a0 100755
--- a/components/3020-swap
+++ b/components/3020-swap
@@ -48,15 +48,15 @@ Swap ()
 	done
 
 	# Remove all auto swap entries
-	if grep -qs "swap swap" /root/etc/fstab.d/swap
+	if grep -qs "swap swap" /root/etc/fstab
 	then
-		grep -v "swap swap" /root/etc/fstab.d/swap > /root/etc/fstab.d/swap.tmp
-		mv /root/etc/fstab.d/swap.tmp /root/etc/fstab.d/swap
+		grep -v "swap swap" /root/etc/fstab > /root/etc/fstab.tmp
+		mv /root/etc/fstab.tmp /root/etc/fstab
 	fi
 
 	# Add new swap entries
 	for _DEVICE in ${_SWAP_DEVICES}
 	do
-		echo "${_DEVICE} swap swap defaults 0 0" >> /root/etc/fstab.d/swap
+		echo "${_DEVICE} swap swap defaults 0 0" >> /root/etc/fstab
 	done
 }
diff --git a/components/9990-fstab.sh b/components/9990-fstab.sh
index 5ae9546..a8d9123 100755
--- a/components/9990-fstab.sh
+++ b/components/9990-fstab.sh
@@ -15,14 +15,14 @@ Fstab ()
 
 	log_begin_msg "Configuring fstab"
 
-	if ! grep -qs  "^${UNIONTYPE}" /root/etc/fstab.d/live
+	if ! grep -qs  "^${UNIONTYPE}" /root/etc/fstab
 	then
-		echo "${UNIONTYPE} / ${UNIONTYPE} rw 0 0" >> /root/etc/fstab.d/live
+		echo "${UNIONTYPE} / ${UNIONTYPE} rw 0 0" >> /root/etc/fstab
 	fi
 
-	if ! grep -qs "^tmpfs /tmp" /root/etc/fstab.d/live
+	if ! grep -qs "^tmpfs /tmp" /root/etc/fstab
 	then
-		echo "tmpfs /tmp tmpfs nosuid,nodev 0 0" >> /root/etc/fstab.d/live
+		echo "tmpfs /tmp tmpfs nosuid,nodev 0 0" >> /root/etc/fstab
 	fi
 
 	log_end_msg
-- 
2.1.3

